### ISSUE TITLE
[docs][guide] New guide on deploying to DigitalOcean

### ISSUE THREAD

**Issue by AishaBlake** [gatsbyjs/gatsby#24549]
## Summary

We've had a few folks ask for documentation around deploying to DigitalOcean. The conversation started in #20691. Rather than a more general "other" category, I think it makes sense to continue to respond to requests for specific docs now that we've got many of the most popular options covered.

Some recommended topics to cover:

- DigitalOcean Droplets
- SSL
- Gotchas like making sure the [`public` directory is in the right place](https://github.com/gatsbyjs/gatsby/issues/5484)

### Motivation

Anyone who's already using DigitalOcean and wants to try Gatsby or add a Gatsby site to their existing server is likely to benefit from this doc.

### Draft the doc

- [ ] Write the doc, following the format listed in these resources:
  - [Overview on contributing to documentation](https://www.gatsbyjs.org/contributing/docs-contributions/)
  - [Docs Templates](https://www.gatsbyjs.org/contributing/docs-templates/)
  - [Example of a similar article](https://www.gatsbyjs.org/docs/deploying-to-netlify/)
  - [Potential resource: blog post](https://delvalle.dev/posts/serving-gatsby-from-digital-ocean-nginx/serving-gatsby-from-digital-ocean-nginx/)
- [ ] Add the article to the [docs sidebar](https://github.com/gatsbyjs/gatsby/blob/master/www/src/data/sidebars/doc-links.yaml) under the Deploying and Hosting section.

### Open a pull request

- [ ] Open a pull request with your work including the words "closes #24549" in the pull request description

**Comment by duncanjbain** [comment]
I think this would be really useful, it’s something that I’ve done quite recently and I’ve had to spend a bit of time trawling various blog posts whenever I’ve hit roadblocks. This is mainly because I’ve only ever deployed Gatsby on Netlify which is ridiculously easy!

I was thinking of writing something for my own personal site but I’m going to redirect my efforts and hopefully get something that’s worth putting in a PR for this issue.

I imagine the majority of the guide will be host agnostic, are there any thoughts on having another piece of documentation that relates to hosting on Linux in general?

**Comment by mistryvatsal** [comment]
Hi @AishaBlake ,

I have created a PR #24652 to close this issue. :smiley:
Waiting for you review.

Thanks,
[Vatsal](https://mistryvatsal.github.io)

**Comment by duncanjbain** [comment]
Hi @AishaBlake and @mistryvatsal , 

I've just created a PR (#24762) to close this issue. I only noticed that you had created one  @mistryvatsal as I was finishing up.

Mine is slightly different, one of them being that it uses nginx's server blocks for the site and  uses the latest Ubuntu LTS version. I'll leave it up for now to see if these differences might be worthwhile having in the tutorial if it's added to the Gatsby docs.
